### PR TITLE
By default, the scraper should only scrape new terms

### PR DESCRIPTION
**Pull request by hankewyczz**
# Purpose

_In 2-3 sentences - what does this code actually do, and why?_

Our scrapers used to be divided:
- Scraping scrapes class data
- Updating scrapes section data

Now, with #190, that line has been blurred. Once a term is scraped initially, the updater will automatically scrape any new classes _for that term_.

Therefore, there's no need to scrape a term more than once anymore (unless the class titles, descriptions, etc. change).

So, we should default our scrapers to only scrape __NEW__ terms. Of course, this can be overridden by several env. variables.

# Reviewers

Primary reviewer:

- Pick one primary reviewer
  - The team member with the most relevant knowledge about the code area this PR touches
  - **NOT** an author of the PR
  - If the primary reviewer is the project lead, _select two primary reviewers_
    - Goal: facilitate knowledge transfer to other team members
  - Primary reviewers **are required to approve the PR** before it can be merged

**Primary**:

Use the **"Reviewers"** feature in Github

Secondary reviewers:

- Pick as many as appropriate — anyone who would benefit from reading the PR
  - Tag them using their Github usernames below

**Secondary**:

### CODE REVIEW

**Review by sebwittr — APPROVED** [review]
Changes look good to me!

File: scrapers/main.ts [hunk 1]
```diff
@@ -14,47 +14,45 @@ import "colors";
...
+      return termIds.filter((t) => !termIdsWithData.has(t));
+    }
   }
 
   async main(): Promise<void> {
     const start = Date.now();
     // Get the TermInfo information from Banner
     const allTermInfos = await bannerv9parser.getAllTermInfos();
-    const termsToScrape = this.getTermIdsToScrape(
+    const termsToScrape = await this.getTermIdsToScrape(
```
Comment (Lucas-Dunker) [inline scrapers/main.ts:55]: Slightly unrelated, but is there ever even a time where we should use `await` for async calls like this? Was `await` not being in any of these just an optimization oversight?

File: scrapers/classes/tests/classScraper.test.ts [hunk 2]
```diff
@@ -1,36 +1,62 @@
-import scraper, { NUMBER_OF_TERMS_TO_UPDATE } from "../../main";
+import scraper from "../../main";
+import prisma from "../../../services/prisma";
 
 describe("getTermsIds", () => {
   beforeEach(() => {
-    process.env.TERMS_TO_SCRAPE = "";
+    delete process.env.TERMS_TO_SCRAPE;
```
Comment (Lucas-Dunker) [inline scrapers/classes/tests/classScraper.test.ts:6]: Why do we delete these environment variables for the tests?

File: scrapers/main.ts [hunk 3]
```diff
@@ -14,47 +14,45 @@ import "colors";
...
       macros.log("Scraping using user-provided TERMS_TO_SCRAPE");
       return terms;
-    }
-
-    const rawNumTerms = Number.parseInt(process.env.NUMBER_OF_TERMS);
-    const numTerms = isNaN(rawNumTerms)
-      ? NUMBER_OF_TERMS_TO_UPDATE
-      : rawNumTerms;
+    } else if (!isNaN(numOfTermsStr)) {
+      return termIds.slice(0, numOfTermsStr);
```
Comment (Lucas-Dunker) [inline scrapers/main.ts:40]: In what case would this code be executed?

**Review by Lucas-Dunker — COMMENTED** [review]
So, in summary, does this refactor make the default case of scraping that only new terms are scraped, but if environment variables are found, it scrapes for those terms instead?

File: scrapers/main.ts (hunk 1 above)
Comment (hankewyczz) [reply scrapers/main.ts:55]: This wasn't an `async` call before - it is now

File: scrapers/main.ts (hunk 3 above)
Comment (hankewyczz) [reply scrapers/main.ts:40]: If `numOfTermsStr` exists and is a number - ie. if someone set the environment variable `NUMBER_OF_TERMS` to a number.

**Review by Lucas-Dunker — APPROVED** [review]

**Pull request merged**